src/common/host_device_vector.cc

```diff
@@ -159,6 +159,16 @@ bool HostDeviceVector<T>::DeviceCanWrite() const {
 template <typename T>
 void HostDeviceVector<T>::SetDevice(int device) const {}
 
+}  // namespace xgboost
+
+namespace {
+
+template<int> struct DummyType { };
```

Comment:
What is this?

Reply:
This was for defining HostDeviceVector for size_t, whose length differs depending on system. But it was addressed in another PR. Will remove.